client/layout/guided-tours/config-elements/step.js

```diff
@@ -111,6 +111,9 @@ export default class Step extends Component {
 	}
 
 	componentWillReceiveProps( nextProps, nextContext ) {
+		if ( this.props.wait !== nextProps.wait ) {
+			this.waitPromise = null;
```

Comment:
you may need a way to cancel the old promise to prevent it from resolving here... Even though you're nulling out the handle to the promise, that doesn't stop it from possibly resolving and running it's callbacks.

Reply:
same goes if the component unmounts while the promise is pending.